src/front/listener.rs

```diff
@@ -25,6 +26,29 @@ pub struct ListenerConfig {
     pub timeout: u64,
 }
 
+/// Specifies metadata associated with a connection, if any.
+#[derive(Copy, Clone, Debug)]
+pub enum ConnectionMetadata {
+    NoMetadata,
+}
+
+/// Represents a connection to a single client. Contains a stream, used for communication with the
+/// client, and some metadata associated with the connection that might be useful elsewhere (i.e.
+/// authentication, etc).
+pub struct Connection {
+    pub stream: Box<dyn ReadWrite + Send>,
+    pub metadata: ConnectionMetadata,
+}
+
+impl fmt::Debug for Connection {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        // NOTE: we omit the field `stream` because it doesn't implement `Debug`.
+        f.debug_struct("Connection")
+            .field("metadata", &self.metadata)
+            .finish()
+    }
+}
```

Comment:
This is perfectly correct, but you can do it more easily with `#[derivative(...)]` which allows you to ignore some fields from `Debug` - you can look at `BackEndHandler` as an example.

Reply:
TIL! Will change.